src/mbgl/map/transform_state.hpp

```diff
@@ -74,13 +77,17 @@ class TransformState {
     double zoomScale(double zoom) const;
     double scaleZoom(double scale) const;
 
+    constexpr explicit operator bool() const {
+        return size && (scale >= min_scale && scale <= max_scale);
+    }
+
 private:
     bool rotatedNorth() const;
     void constrain(double& scale, double& x, double& y) const;
 
     // Limit the amount of zooming possible on the map.
-    double min_scale = std::pow(2, 0);
-    double max_scale = std::pow(2, 20);
+    double min_scale = 1.0; // 2^0
+    double max_scale = 1048576.0; // 2^20
```

Comment:
These values are known and `std::pow` is not `constexpr`, causing the `TransformState` ctor to not be `constexpr`-able too.